pkg/cmd/pulumi/templates/org.go

```diff
@@ -48,17 +48,22 @@ func (s *Source) getOrgTemplates(
 		return
 	}
 
-	b, err := cmdBackend.CurrentBackend(ctx, ws, cmdBackend.DefaultLoginManager, project, display.Options{
-		Color:         cmdutil.GetGlobalColorization(),
-		IsInteractive: interactive,
-	})
+	url, err := pkgWorkspace.GetCurrentCloudURL(ws, env.Global(), project)
+	b, err := cmdBackend.DefaultLoginManager.Current(ctx, ws, cmdutil.Diag(), url, project, false)
 	if err != nil {
 		if !errors.Is(err, backend.MissingEnvVarForNonInteractiveError{}) {
 			s.addError(fmt.Errorf("could not get the current backend: %w", err))
 		}
 		logging.Infof("could not get a backend for org templates")
 		return
 	}
+
+	// Attempt to retrieve the current user
+	if _, _, _, err := b.CurrentUser(); err != nil {
+		logging.Infof("could not get the current user for %s: %s", url, err)
+		return // No current user - so don't proceed
```

Comment:
@Frassle I rejiggered things so that we can check `b.CurrentUser`. What I don't love here is that this might swallow a different error here.

The problem is that we don't have a "ErrRequiresLogin" or similar. The error comes back to us from https://github.com/pulumi/pulumi/blob/fee07fa261db7088bc18319c0f00d94bc1048537/pkg/backend/httpstate/client/api.go#L338

Do we need to return a distinct error from the api client, look for that in httpstate and then return it from there as a `backend.ErrRequiresLogin` or similar?

Reply:
Feels ok to change that to a specific error type. Could also just add something to the backend interface to ask for the current `workspace.Account` which will be empty if we didn't log in ok.